config/samples/beats/metricbeat-kubernetes.yaml

```diff
@@ -0,0 +1,377 @@
+          readOnly: true
+        - name: dockersock
+          mountPath: /var/run/docker.sock
+        - name: proc
+          mountPath: /hostfs/proc
+          readOnly: true
+        - name: cgroup
+          mountPath: /hostfs/sys/fs/cgroup
+          readOnly: true
+        - name: es-certs
+          mountPath: /mnt/elastic/tls.crt
+          readOnly: true
+          subPath: tls.crt
+        - name: kb-certs
+          mountPath:  /mnt/kibana/ca.crt
+          readOnly: true
+          subPath: ca.crt
+      volumes:
+      - name: proc
+        hostPath:
+          path: /proc
+      - name: cgroup
+        hostPath:
+          path: /sys/fs/cgroup
+      - name: dockersock
+        hostPath:
+          path: /var/run/docker.sock
+      - name: config
+        configMap:
+          defaultMode: 0600
+          name: metricbeat-daemonset-config
+      - name: modules
+        configMap:
+          defaultMode: 0600
+          name: metricbeat-daemonset-modules
+      - name: data
+        hostPath:
+          path: /var/lib/metricbeat-data
+          type: DirectoryOrCreate
+      - name: es-certs
+        secret:
+          secretName: monitor-es-http-certs-public
+      - name: kb-certs
+        secret:
+          secretName: monitor-kb-http-certs-public
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metricbeat-deployment-config
+  namespace: beats
+  labels:
+    k8s-app: metricbeat
+data:
+  metricbeat.yml: |-
+    metricbeat.config.modules:
+      # Mounted `metricbeat-daemonset-modules` configmap:
+      path: ${path.config}/modules.d/*.yml
+      # Reload module configs as they change:
+      reload.enabled: false
+
+    processors:
+      - add_cloud_metadata:
+
+    output.elasticsearch:
+      hosts: ['http://${ELASTICSEARCH_HOST:elasticsearch}:${ELASTICSEARCH_PORT:9200}']
+      username: ${ELASTICSEARCH_USERNAME}
+      password: ${ELASTICSEARCH_PASSWORD}
+      ssl.certificate_authorities:
+      - /mnt/elastic/tls.crt
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metricbeat-deployment-modules
+  namespace: beats
+  labels:
+    k8s-app: metricbeat
+data:
+  # This module requires `kube-state-metrics` up and running under `kube-system` namespace
+  kubernetes.yml: |-
+    - module: kubernetes
+      metricsets:
+        - state_node
+        - state_deployment
+        - state_replicaset
+        - state_pod
+        - state_container
+        # Uncomment this to get k8s events:
+        #- event
+      period: 10s
+      host: ${NODE_NAME}
+      hosts: ["kube-state-metrics:8080"]
+---
+# Deploy singleton instance in the whole cluster for some unique data sources, like kube-state-metrics
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: metricbeat
+  namespace: beats
+  labels:
+    k8s-app: metricbeat
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: metricbeat
+    spec:
+      serviceAccountName: metricbeat
+      hostNetwork: true
```

Comment:
Why does the deployment need access to the host network?

Reply:
Looks like it is to report the actual hostname of the machine and not the name of the metricbeat container when fetching node-wide metrics from kube-state-metrics (https://github.com/elastic/beats/pull/7463).
It seems to me that the downward API could be used instead?

Reply:
I'm wondering if something prevents the use of the downward API to inject the NODE_NAME and set it as the [beat name](https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-general-options.html#_name) (used instead of the hostname when defined).

cc @exekias

Reply:
We indeed use this to report host metadata, by using the [add_host_metadata](https://www.elastic.co/guide/en/beats/metricbeat/current/add-host-metadata.html) processor. We could consider copying NODE_NAME to `host.name`, but would lose the rest